### ISSUE TITLE
support env vars for parity with other agents 

### ISSUE THREAD

**Issue by novas0x2a** [newrelic/go-agent#104]
Some of the other language agents support environment-based config (e.g. [java](https://docs.newrelic.com/docs/agents/java-agent/configuration/java-agent-configuration-config-file#Environment_Variables), [node](https://docs.newrelic.com/docs/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration#environment)), would it be possible for this lib to support them too, probably by loading them in NewConfig and only falling back to the current defaults if the vars are unset? Thanks!

**Comment by purple4reina** [comment]
@novas0x2a Thanks a ton for sharing this issue with us! We think this is a great idea. We're in the process now of planning for a 3.0 release of which this would be a part.

Our idea is to simplify the creation of the Application object. Instead of needing to create a Config, then an Application from the Config, you'd use one method to create the Application and pass it a set of options much like we've done in [our telemetry-sdk](https://godoc.org/github.com/newrelic/newrelic-telemetry-sdk-go/telemetry#NewHarvester). It would look something like this:

```go
app, err := newrelic.NewApplication("MyAppName", "MyLicenseKey", newrelic.ConfigDistributedTracingEnabled)
```

One such option we could have would allow reading from environment variables. So, in your case, you'd do something like this:

```go
app, err := newrelic.NewApplication("MyAppName", "MyLicenseKey", newrelic.ConfigFromEnvVar)
```

Notice that the agent wouldn't read settings from environment variables by default, but you could easily configure it to do so.

What are your thoughts? Would this type of system fit your needs?

**Comment by purple4reina** [comment]
Hi @novas0x2a!

We've recently published some proposed changes to the agent for a 3.0 major version release https://github.com/newrelic/go-agent/issues/106. We're wondering what you think of these changes and if they will fit your needs.

**Comment by novas0x2a** [comment]
This seems fine, yep! They're not listed in the doc, but as long as the env vars match up with the other clients it looks like this would work fine for us (we set the env vars globally, so we're very aware of mismatches like node's `NEW_RELIC_ERROR_COLLECTOR_IGNORE_ERROR_CODES` vs java's `NEW_RELIC_ERROR_COLLECTOR_IGNORE_STATUS_CODES`)

**Comment by purple4reina** [comment]
Hey @novas0x2a!

Thanks again for opening this issue! We just released v3.0.0 of the agent which has support for setting configuration via environment variables. Check it out and let us know what you think! If there's any env vars that you'd like added, just let us know!
